volatility/framework/symbols/windows/pdb.py

```diff
@@ -0,0 +1,291 @@
+                    potential_output_filename))
+                continue
+            finally:
+                # If something else failed, removed the symbol file so we don't pick it up in the future
+                if not data_written and os.path.exists(potential_output_filename):
+                    os.remove(potential_output_filename)
+                # Clear out all the temporary file if we constructed one
+                for filename in tmp_files:
+                    try:
+                        os.remove(filename)
+                    except PermissionError:
+                        vollog.warning("Temporary file could not be removed: {}".format(filename))
+        else:
+            vollog.warning("Cannot write downloaded symbols, please add the appropriate symbols"
+                           " or add/modify a symbols directory that is writable")
+
+    @classmethod
+    def pdbname_scan(cls,
+                     ctx: interfaces.context.ContextInterface,
+                     layer_name: str,
+                     page_size: int,
+                     pdb_names: List[bytes],
+                     progress_callback: constants.ProgressCallback = None,
+                     start: Optional[int] = None,
+                     end: Optional[int] = None) -> Generator[Dict[str, Optional[Union[bytes, str, int]]], None, None]:
+        """Scans through `layer_name` at `ctx` looking for RSDS headers that
+        indicate one of four common pdb kernel names (as listed in
+        `self.pdb_names`) and returns the tuple (GUID, age, pdb_name,
+        signature_offset, mz_offset)
+
+        .. note:: This is automagical and therefore not guaranteed to provide correct results.
+
+        The UI should always provide the user an opportunity to specify the
+        appropriate types and PDB values themselves
+        """
+        min_pfn = 0
+
+        if start is None:
+            start = ctx.layers[layer_name].minimum_address
+        if end is None:
+            end = ctx.layers[layer_name].maximum_address
+
+        for (GUID, age, pdb_name,
+             signature_offset) in ctx.layers[layer_name].scan(ctx,
+                                                              PdbSignatureScanner(pdb_names),
+                                                              progress_callback = progress_callback,
+                                                              sections = [(start, end - start)]):
+            mz_offset = None
+            sig_pfn = signature_offset // page_size
+
+            for i in range(sig_pfn, min_pfn, -1):
+                if not ctx.layers[layer_name].is_valid(i * page_size, 2):
+                    break
+
+                data = ctx.layers[layer_name].read(i * page_size, 2)
+                if data == b'MZ':
+                    mz_offset = i * page_size
+                    break
+            min_pfn = sig_pfn
+
+            yield {
```

Comment:
@ikelos should this check if mz_offset == None before yielding? Otherwise it seems like it would yield with a None value for it.

Reply:
Nevermind, I see where a None mz_offset is specifically checked in method_fixed_mapping